File: Practical-2/Practical-2.cpp
//  Author: Anish Tilloo
//  Roll No. : 34
//  Program: Bubble Sort

#include <bits/stdc++.h>
using namespace std;

// Bubble Sort Algorithm
void bubbleSort(int n, int arr[]){
    for (int i = 0; i < n; i++)
    {
        for (int  j = 0; j < n - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                // swaping if the number is greater than the next one
                swap(arr[j], arr[j + 1]);
            }
        }
    }
}

void inputArray( int arr[], int arraySize){
    cout << "Enter the elements of the array one by one " << endl;
    for (int i = 0; i < arraySize; i++)
    {
        cin >> arr[i];
    }
}

void printArray(int arr[], int arraySize){
    for (int i = 0; i < arraySize; i++)
    {
        cout << arr[i] << " ";
    }   
}

int main(){
    int n;
    cout << "No. of elements in the array: " << endl;
    cin >> n;
    int arr[n];
    inputArray(arr, n); 
    bubbleSort(n, arr);
    cout << "Sorted Array: " << endl;
    printArray(arr, n);
    return 0;
}

File: Practical-2/Practical_2.py
import array

# Bubble Sort Algorithm
def bubbleSort(arr, n):
    for i in range(0, n):
        for j in range(0, n - 1):
            if arr[j] > arr[j + 1]:
                # swaping if the number is greater than the next one
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

    # printing the sorted array
    print("Sorted array: ")
    for i in arr:
         print(i)
            
# Array
# i is the iterating variable for the array
arr = array.array("i")

# inputing the size of array
n = int(input("Enter the size of the array: "))

print("\n")

# inputing the array
print("Enter the array elements: ")
for i in range(0, n):
    element = int(input())
 
    arr.append(element)

print("\n")
bubbleSort(arr, n)



                
            
    